### PR TITLE
fixes #10940

### LINKED ISSUE #10940: Destructor regression for tuples unpacking
I will provide test and PR

### PR DESCRIPTION
**Pull request by cooldome**
included test currently fails with internal compiler error: copy assignment is not available for tuple (MyVal , MyVal)

### CODE REVIEW

File: compiler/semstmts.nim [hunk 1]
```diff
@@ -2118,7 +2118,7 @@ proc semStmtList(c: PContext, n: PNode, flags: TExprFlags): PNode =
             continue
 
         let verdict = semConstExpr(c, n[i])
-        if verdict.intVal == 0:
+        if verdict == nil or verdict.kind != nkIntLit or verdict.intVal == 0:
```
Comment (cooldome) [inline compiler/semstmts.nim]: I have found that megatest doesn't compile with debug nim (verdict is not integer node is some of the tests hence I had to fix it)

File: compiler/liftdestructors.nim [hunk 2]
```diff
@@ -600,7 +600,7 @@ proc createTypeBoundOps*(c: PContext; orig: PType; info: TLineInfo) =
   ## In the semantic pass this is called in strategic places
   ## to ensure we lift assignment, destructors and moves properly.
   ## The later 'injectdestructors' pass depends on it.
-  if orig == nil or {tfCheckedForDestructor, tfHasMeta} * orig.flags != {}: return
+  if orig == nil or {tfCheckedForDestructor, tfHasMeta} * orig.skipTypes({tyAlias}).flags != {}: return
```
Comment (cooldome) [inline compiler/liftdestructors.nim]: This one is extremely weird but helps fixing failing test. By the looks of it, i have digest collision in sighashes in code below. orig type is `distinct float` and canon type I get is `alias proxy`, very weird. Skipping tyAlias makes sure tyProxy doesn't get into sighashes table below.

This happens only in megatest and doesn't happen if you run tests individually.

orig type is `type USD = distinct float` from tests\concepts\tconcepts_issues.nim
canon type is `type VectorSpace[K] = concept x, y` from tests\concepts\tconcepts.nim

**Comment by cooldome** [comment]
please review if tests will pass

File: compiler/liftdestructors.nim (hunk 2 above)
Comment (Araq) [reply compiler/liftdestructors.nim]: Yeah megatest triggers quite some concept-related problems. Acceptable for now.

**Pull request merged**